### ISSUE TITLE
Monthly aggregate / average (feature request)

### ISSUE THREAD

**Issue by dorinpirvu** [KensoBI/spc-chart#35]
![image](https://github.com/KensoBI/spc-panel/assets/127939087/d185b3a0-debe-400f-b336-6a6cd0f5c9d6)
Hello, 
It could be useful to be able to display more data for example like in the attached picture: it displays the monthly average together with each individual sample (plus target and limits)

**Comment by mrtomeq** [comment]
We should be able to accomplish it by implementing a way to switch between a line and a point on each axis. We will work on it. Thanks for your suggestion!

![image](https://github.com/KensoBI/spc-panel/assets/8570009/44cdfdbd-6611-4937-8efa-e827ee1d7cb9)

**Comment by dorinpirvu** [comment]
Thanks for the comment!
Also maybe take into consideration the possibility that there could be a different number of samples each month (or other time period aggregation / variable number of samples in a bin in general)

**Comment by mrtomeq** [comment]
We use a constant sample size because of the type of calculations we perform (control limits), which require a constant bin size. I don't think we could use the same equations with a variable sample size. However, we could calculate simple things like averages, standard deviation, min, max, etc. with a variable sample size.
How would we decide on which value to group measurements by (how to construct a bin)? Maybe a text box or a drop-down menu showing all available columns, and the user can select from the menu?

**Comment by dorinpirvu** [comment]
Yes, good points, that's exactly why I was asking, I also had trouble because of variable bin size; the only workaround to get what I wanted was to have two different queries in Grafana, one for the individual samples and one that calculates the weighted average per month.
So, one idea is to have month (or some time period - day, week, year - that is relevant to the user and sample population) as the grouping of measurements; in our case it was desired to see an average value per month (and per product type). There could be another workaround-idea that could prove useful, which we used in excel before: assume a max value of samples per time period and create bins with that size (for example we noticed we had max 20 batches of the same product each month, so in excel we had max 20 series that would get plotted on a graph (whether they were empty or not)).

**Comment by mrtomeq** [comment]
After reviewing this, we have decided not to implement this as a built-in feature in the panel.

The workaround you're already using - creating separate Grafana queries for individual samples and monthly aggregates - is actually the recommended approach. This aligns with Grafana's design philosophy and provides you with more flexibility:

How to achieve this:

1. Create one query for your individual samples
2. Create a second query that calculates monthly averages (using your data source's aggregation functions)
3. Add both queries to the same SPC Chart panel
4. Use the panel's styling options to differentiate between them:
   - Set one series to display as Points (drawStyle)
   - Set the other as a Line
   - Customize colors, line widths, etc. per series

This approach:

- Gives you full control over how each series is calculated
- Keeps the panel focused on SPC calculations rather than duplicating Grafana's built-in query capabilities
- Allows for more flexibility (you can aggregate by any time period, not just monthly)
- Works with any data source

The panel already supports displaying multiple series together with control limits, so this multi-query approach should cover your use case effectively.

Closing this as won't implement since the functionality is already achievable through Grafana's native multi-query capabilities.
